protoc-gen-swagger/genswagger/template.go

```diff
@@ -1448,6 +1458,188 @@ func processExtensions(inputExts map[string]*structpb.Value) ([]extension, error
+				"int64",
+				"float",
+				"float32",
+				"float64",
+				"complex64",
+				"complex128",
+				"double",
+				"byte",
+				"rune",
+				"uintptr":
+				return nil
+			}
+			return fmt.Errorf("the provided format %q is not a valid extension of the type %q", format, headerType)
+		}
+		return nil
+	case "integer":
+		if format != "" {
+			switch format {
+			case
+				"uint",
+				"uint8",
+				"uint16",
+				"uint32",
+				"uint64",
+				"int",
+				"int8",
+				"int16",
+				"int32",
+				"int64":
+				return nil
+			}
+			return fmt.Errorf("the provided format %q is not a valid extension of the type %q", format, headerType)
+		}
+		return nil
+	case "boolean":
+		return nil
+	}
+	return fmt.Errorf("the provided header type %q is not supported", headerType)
+}
+
+func validateDefaultValueTypeAndFormat(headerType string, defaultValue string, format string) error {
+	switch headerType {
+	case "string":
+		if !isQuotedString(defaultValue) {
+			return fmt.Errorf("the provided default value %q does not match provider type %q, or is not properly quoted with escaped quotations", defaultValue, headerType)
+		}
+		if format != "" {
+			switch format {
+			case "date-time":
+				unquoteTime := strings.Trim(defaultValue, `"`)
+				_, err := time.Parse(time.RFC3339, unquoteTime)
+				if err != nil {
+					return fmt.Errorf("the provided default value %q is not a valid RFC3339 date-time string", defaultValue)
+				}
+			case "date":
+				const (
+					layoutRFC3339Date = "2006-01-02"
+				)
+				unquoteDate := strings.Trim(defaultValue, `"`)
+				_, err := time.Parse(layoutRFC3339Date, unquoteDate)
+				if err != nil {
+					return fmt.Errorf("the provided default value %q is not a valid RFC3339 date-time string", defaultValue)
+				}
+			}
+		}
+	case "number":
+		err := isJSONNumber(defaultValue, headerType)
+		if err != nil {
+			return err
+		}
+	case "integer":
+		if format != "" {
+			switch format {
+			case "int32":
+				_, err := strconv.ParseInt(defaultValue, 0, 32)
+				if err != nil {
+					return fmt.Errorf("the provided default value %q does not match provided format %q", defaultValue, format)
+				}
+			case "uint32":
+				_, err := strconv.ParseUint(defaultValue, 0, 32)
+				if err != nil {
+					return fmt.Errorf("the provided default value %q does not match provided format %q", defaultValue, format)
+				}
+			case "int64":
+				_, err := strconv.ParseInt(defaultValue, 0, 64)
+				if err != nil {
+					return fmt.Errorf("the provided default value %q does not match provided format %q", defaultValue, format)
+				}
+			case "uint64":
+				_, err := strconv.ParseUint(defaultValue, 0, 64)
+				if err != nil {
+					return fmt.Errorf("the provided default value %q does not match provided format %q", defaultValue, format)
+				}
+			}
+		}
+
+		_, err := strconv.ParseInt(defaultValue, 0, 64)
+		if err != nil {
+			return fmt.Errorf("the provided default value %q does not match provided type %q", defaultValue, headerType)
+		}
```

Comment:
I think we need to rewrite this logic a little, since this will fail on a `format: "uint64"` of sufficient size. It will go into the uint64 parse, then fall back to this generic parse and fail. I suggest adding a test case for a default with math.MaxUint64 as the value (and appropriate type and format set, of course). We'll need to do this generic "format isn't set" parsing in an else, though I think we can write it neater with a single switch inside this case:

```suggestion
        switch format {
        case "int32":
            _, err := strconv.ParseInt(defaultValue, 0, 32)
            if err != nil {
                return fmt.Errorf("the provided default value %q does not match provided format %q", defaultValue, format)
            }
        case "uint32":
            _, err := strconv.ParseUint(defaultValue, 0, 32)
            if err != nil {
                return fmt.Errorf("the provided default value %q does not match provided format %q", defaultValue, format)
            }
        case "uint64":
            _, err := strconv.ParseUint(defaultValue, 0, 64)
            if err != nil {
                return fmt.Errorf("the provided default value %q does not match provided format %q", defaultValue, format)
            }
        case "int64":
            fallthrough
        default:
            _, err := strconv.ParseInt(defaultValue, 0, 64)
            if err != nil {
                return fmt.Errorf("the provided default value %q does not match provided type %q", defaultValue, headerType)
            }
        }
```